File: 백준/1260_DFS와BFS.cpp
#include<iostream>
#include<cstring>
#include<algorithm>
#include<vector>
#include<queue>
using namespace std;
int n, v, m, a, b;
vector<int> arr[1005];
int check[1005];
queue<int> q;
void dfs(int here) {
	check[here] = 1;
	printf("%d ", here);
	for (int i = 0; i < arr[here].size(); i++) {
		int next = arr[here][i];
		if (!check[next]) {
			check[next] = 1;
			dfs(next);
		}
	}
}
void bfs() {
	q.push(v);
	while (!q.empty()) {
		int here = q.front();
		printf("%d ", here);
		check[here] = 1;
		q.pop();
		for (int i = 0; i < arr[here].size(); i++) {
			int next = arr[here][i];
			if (!check[next]) {
				check[next] = 1;
				q.push(next);
			}
		}
	}
}
int main(void) {
	scanf("%d%d%d", &n, &m, &v);
	for (int i = 1; i <= m; i++) {
		scanf("%d%d", &a, &b);
		arr[a].push_back(b);
		arr[b].push_back(a);
	}
	for (int i = 1; i <= n; i++)sort(arr[i].begin(), arr[i].end());
	dfs(v);
	memset(check, 0, sizeof(check));
	printf("\n");
	bfs();
	return 0;
}

File: 백준/1260_DFS와BFS.kt
import java.util.*;

var n : Int = 0
var m : Int = 0
var v : Int = 0
val graph = Array(1005) { mutableListOf<Int>() }
val check = Array(1005) { false }
val answer = mutableListOf<Int>()

fun main(args: Array<String>) = with(Scanner(System.`in`)) {
    var a : Int
    var b : Int

    n = nextInt()
    m = nextInt()
    v = nextInt()

    for(i: Int in 1..m) { // 그래프 생성
        a = nextInt()
        b = nextInt()

        graph[a].add(b)
        graph[b].add(a)
        graph[a].sort()
        graph[b].sort()
    }

    dfs(v);
    for(i: Int in 0 until answer.size) {
        System.out.print(answer[i])
        System.out.print(" ")
    }
    System.out.println()
    answer.clear()
    for(i in 0 until check.size) {
        check[i] = false
    }
    bfs()
    for(i: Int in 0 until answer.size) {
        System.out.print(answer[i])
        System.out.print(" ")
    }
}

fun dfs(here : Int) {
    answer.add(here)
    check[here] = true
    for(i : Int in 0 until  graph[here].size) {
        val next : Int = graph[here][i]
        if(!check[next]) {
            check[next] = true
            dfs(next)
        }
    }
}

fun bfs() {
    val queue: Queue<Int> = LinkedList<Int>()
    queue.add(v)
    while(!queue.isEmpty()) {
        val here = queue.poll()
        answer.add(here)
        check[here] = true
        for(i in 0 until graph[here].size) {
            val next = graph[here][i]
            if(!check[next]) {
                check[next] = true
                queue.add(next)
            }
        }
    }
}